crates/krun-guest/src/net.rs

```diff
@@ -18,7 +20,47 @@ pub fn configure_network() -> Result<()> {
         sethostname(hostname.as_bytes()).context("Failed to set hostname")?;
     }
 
-    let output = Command::new("/sbin/dhclient")
+    let dhcpcd_path = find_in_path("dhcpcd").context("Failed to check existence of `dhcpcd`")?;
+    let dhcpcd_path = if let Some(dhcpcd_path) = dhcpcd_path {
+        Some(dhcpcd_path)
+    } else {
+        find_executable("/sbin/dhcpcd").context("Failed to check existence of `/sbin/dhcpcd`")?
+    };
+    if let Some(dhcpcd_path) = dhcpcd_path {
+        let output = Command::new(dhcpcd_path)
+            .args(["-M", "--nodev", "eth0"])
```

Comment:
`--nodev` is compulsory here. Otherwise no matter what we pass, `dhcpcd` just exits and/or refuses to work.